### PR TITLE
Fix executing with JDK headless

### PR DESCRIPTION
**Pull request by ItaloYeltsin**
## Description:
Describe your changes in details. 

### Related Issue:
- #27 
## Motivation and Context:

Solves the problem with deploying totalcross java components from api to tcz when using Java Headless.

## Benefited Devices:
 - Device: arm devices.
 - OS: for building yocto images.

## How Has This Been Tested?
 - build passed by test made by OS System.

## Tested Devices:
 - Device: Linux ARM devices

### CODE REVIEW

**Comment by otavio** [comment]
Tested on Yocto Project and it worked just fine.

Fixes: #27 

LGTM.

File: TotalCrossSDK/src/main/java/totalcross/ui/image/Image.java [hunk 1]
```diff
@@ -1785,26 +1810,46 @@ public ImageLoader(byte[] input, int len) {
...
-            throw new RuntimeException();
+        ImageInputStream stream = ImageIO.createImageInputStream(new ByteArrayInputStream(input, 0, len));
+        ImageReader reader = ImageIO.getImageReaders(stream).next();
+        reader.setInput(stream);
+
+        if (new String(imgBytes, 0, 3).equals("GIF")) {
+          width = readWord(input, 6);
+          height = readWord(input, 8);
+
+          // int count = reader.getNumImages(true);
```
Comment (otavio) [inline TotalCrossSDK/src/main/java/totalcross/ui/image/Image.java:1821]: dead code.

File: TotalCrossSDK/src/main/java/totalcross/ui/image/Image.java [hunk 2]
```diff
@@ -1785,26 +1810,46 @@ public ImageLoader(byte[] input, int len) {
...
+          try {
+            for (int index = 0;; index++) {
+              BufferedImage frame = reader.read(index);
+
+              Image img = new Image(width, height);
+              img.pixels = convertToPixels(frame, img.pixels, width, height);
+              frames.add(img);
+            }
+          } catch (IndexOutOfBoundsException e) {
+            // e.printStackTrace();
```
Comment (otavio) [inline TotalCrossSDK/src/main/java/totalcross/ui/image/Image.java:1831]: dead code.

**Comment by otavio** [comment]
Can we re-test this? or should we wait?

**Comment by flsobral** [comment]
> It seems you succeed in reducing a lot the code involved in this processing. There are some commented code and it would be nice if it could be removed.

yeah, this new implementation uses only ImageIO, nothing from awt and media toolkit which are not always available on headless environment.
The dead code you mentioned before was intentional, but I wasn't sure about the best place to insert a comment explaining why the loop doesn't have a break condition.
In the end I decided to explain everything inside the catch clause but keeping the code for the break condition inside block comment for clarity.

**Comment by flsobral** [comment]
> Can we re-test this? or should we wait?

Yup, it's done. :)

**Comment by otavio** [comment]
Are you intending to squash the changes? if so, please do and we can run a test here.

**Comment by ItaloYeltsin** [comment]
@otavio waiting for the test! =)

**Pull request merged**